File: integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/Address.java
package io.quarkus.it.panache.reactive;

import jakarta.persistence.Entity;

import io.quarkus.hibernate.reactive.panache.PanacheEntity;

@Entity
public class Address extends PanacheEntity implements Comparable<Address> {

    public String street;

    public Address() {
    }

    public Address(String street) {
        this.street = street;
    }

    @Override
    public int compareTo(Address address) {
        return street.compareTo(address.street);
    }
}

File: integration-tests/hibernate-reactive-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/reactive/kotlin/Address.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntity
import jakarta.persistence.Entity

@Entity
class Address : PanacheEntity, Comparable<Address> {
    companion object : PanacheCompanion<Address>

    var street: String? = null

    constructor() {}
    constructor(street: String) {
        this.street = street
    }

    override operator fun compareTo(address: Address): Int {
        return street!!.compareTo(address.street!!)
    }
}
